src/codeu/chat/server/BroadCastSystem.java

```diff
@@ -0,0 +1,148 @@
+package codeu.chat.server;
+
+import codeu.chat.common.*;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.Uuid;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Created by rsharif on 3/17/17.
+ */
+
+
+public class BroadCastSystem {
+
+    private class ConversationMessageLink {
+
+        Uuid conversationUuid;
+        Message message;
+
+        ConversationMessageLink(Uuid conversationUuid, Message message) {
+            this.conversationUuid = conversationUuid;
+            this.message = message;
+        }
+
+    }
+
+    private ConcurrentHashMap<Integer, List<Connection>> conversationUsers;
```

Comment:
Currently, the messages do not store any information regarding which conversation they are a part of. Conversation only stores a reference to the first message. Each message then references the next message in the chain, similar to a linked list. 

In regards to the first comment, I'm thinking I will switch the Key to be Conversation.Uuid.